pkg/shp/cmd/build/create.go

```diff
@@ -63,16 +66,44 @@ func (c *CreateCommand) Run(params *params.Params, io *genericclioptions.IOStrea
 
 	flags.SanitizeBuildSpec(&b.Spec)
 
-	// print warning with regards to source bundle image being used
-	if b.Spec.Source.BundleContainer != nil && b.Spec.Source.BundleContainer.Image != "" {
-		fmt.Fprintf(io.Out, "Build %q uses a source bundle image, which means source code will be transferred to a container registry. It is advised to use private images to ensure the security of the source code being uploaded.\n", c.name)
+	if b.Spec.Source != nil {
+		if b.Spec.Source.OCIArtifact != nil && b.Spec.Source.OCIArtifact.Image != "" {
+			b.Spec.Source.Type = buildv1beta1.OCIArtifactType
+		} else if b.Spec.Source.Git != nil && b.Spec.Source.Git.URL != "" {
+			b.Spec.Source.Type = buildv1beta1.GitType
+		}
+
+		// print warning with regards to source bundle image being used
+		if b.Spec.Source.OCIArtifact != nil && b.Spec.Source.OCIArtifact.Image != "" {
+			fmt.Fprintf(io.Out, "Build %q uses a source bundle image, which means source code will be transferred to a container registry. It is advised to use private images to ensure the security of the source code being uploaded.\n", c.name)
+		}
+	}
+
+	if c.dockerfile != nil && *c.dockerfile != "" {
```

Comment:
nit (not going to block on this) - you can do both checks in golang using `len`, since it is safe to use on nil pointers:

```suggestion
	if len(c.dockerfile) > 0 {
```